apps/nextjs-app/src/lib/withEnv.ts

```diff
@@ -15,12 +15,27 @@ type GetServerSideProps<
   D extends PreviewData = PreviewData,
 > = (context: GetServerSidePropsContext<Q, D>) => Promise<GetServerSidePropsResult<P>>;
 
+async function fetchPublicConfig(context?: GetServerSidePropsContext) {
+  try {
+    const protocol = process.env.NODE_ENV === 'development' ? 'http' : 'https';
```

Comment:
you can use axios from `apps/nextjs-app/src/backend/api/rest/axios.ts`
it design to handle internal request